Log file does not exist

```
self = <mongobox.mongobox.MongoBox object at 0x7fd7903af040>
    def _wait_till_started(self):
        attempts = 0
        while True:
            if self.process.poll() is not None:  # the process has terminated
>               with open(self.log_path) as log_file:
E               FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpnpc2_yzi/mongodb.log'
../venv/lib/python3.8/site-packages/mongobox/mongobox.py:90: FileNotFoundError

```

I get this error when I run with mongo 4.2, but it works correctly with 3.6.
It is because mongo 4.2 does not recognize the arguments --smallfiles, and --noprealloc, and the mongo process never starts

--- a/mongobox/mongobox.py
+++ b/mongobox/mongobox.py
@@ -92,8 +92,11 @@ class MongoBox(object):
         while True:
             if self.process.poll() is not None:  # the process has terminated
                 with open(self.log_path) as log_file:
-                    raise SystemExit('MondgoDB failed to start:\n{}\n{}'.format(
-                        ' '.join(self.process_args), log_file.read()))
+                    try:
+                        raise SystemExit('MondgoDB failed to start:\n{}\n{}'.format(
+                            ' '.join(self.process_args), log_file.read()))
+                    except FileNotFoundError:
+                        raise SystemExit('MondgoDB failed to start')
             attempts += 1
             if attempts > START_CHECK_ATTEMPTS:
                 break
@@ -114,7 +117,7 @@ class MongoBox(object):
         if self.process is None or self.process.poll() is not None:
             # the process does not exist anymore
             return
-        
+
         # Not sure if there should be more checks for
         # other platforms.
         if sys.platform == 'darwin':

--- a/mongobox/mongobox.py
+++ b/mongobox/mongobox.py
@@ -18,8 +18,6 @@ DEFAULT_ARGS = [
     "--quiet",
     # disable unused.
     "--nounixsocket",
-    # use a smaller default file size
-    "--smallfiles",
     # journaling on by default in 2.0 and makes it to slow
     # for tests, can causes failures in jenkins
     "--nojournal",
@@ -35,7 +33,8 @@ START_CHECK_ATTEMPTS = 200
 class MongoBox(object):
     def __init__(self, mongod_bin=None, port=None,
                  db_path=None, scripting=False,
-                 prealloc=False, auth=False, storage_engine=None):
+                 prealloc=True, auth=False, storage_engine=None,
+                 small_files=False):
 
         if db_path and os.path.exists(db_path) and os.path.isfile(db_path):
             raise AssertionError('DB path should be a directory, but it is a file.')
@@ -49,6 +48,7 @@ class MongoBox(object):
         self._db_path_is_temporary = not self.db_path
         self.auth = auth
         self.storage_engine = storage_engine
+        self.small_files = small_files
 
         self.process = None
 
@@ -79,6 +79,10 @@ class MongoBox(object):
 
         if not self.prealloc:
             args.append("--noprealloc")
+
+        # use a smaller default file size
+        if self.small_files:
+            args.append("--smallfiles")
 
         self.process_args = args
 

